api/samples/opcode_count.cpp

```diff
@@ -0,0 +1,165 @@
+                               * here won't be used: drreg's slots will be.
+                               */
+                              static_cast<dr_spill_slot_t>(SPILL_SLOT_MAX + 1),
+                              IF_AARCHXX_(SPILL_SLOT_MAX + 1) & global_opcode_count, 1,
+                              DRX_COUNTER_LOCK);
+
+    return DR_EMIT_DEFAULT;
+}
+
+static dr_emit_flags_t
+event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst,
+                      bool for_trace, bool translating, void *user_data)
+{
+    /* By default drmgr enables auto-predication, which predicates all instructions with
+     * the predicate of the current instruction on ARM.
+     * We disable it here because we want to unconditionally execute the following
+     * instrumentation.
+     */
+    drmgr_disable_auto_predication(drcontext, bb);
+    if (!drmgr_is_first_instr(drcontext, inst))
+        return DR_EMIT_DEFAULT;
+
+    drx_insert_counter_update(drcontext, bb, inst,
+                              /* We're using drmgr, so these slots
+                               * here won't be used: drreg's slots will be.
+                               */
+                              static_cast<dr_spill_slot_t>(SPILL_SLOT_MAX + 1),
+                              IF_AARCHXX_(SPILL_SLOT_MAX + 1) & global_total_count, 1,
+                              DRX_COUNTER_LOCK);
+
+    return DR_EMIT_DEFAULT;
+}
+
+DR_EXPORT void
+dr_client_main(client_id_t id, int argc, const char *argv[])
+{
+    /* parse command-line options */
+    if (!droption_parser_t::parse_argv(DROPTION_SCOPE_CLIENT, argc, argv, NULL, NULL))
+        DR_ASSERT(false);
+
+    /* get opcode and check if valid */
+    int valid_opcode = opcode.get_value();
+    if (valid_opcode < OP_FIRST || valid_opcode > OP_LAST) {
+        dr_fprintf(STDERR, "Error: give a valid opcode as a parameter\n");
+        dr_abort();
+    }
+
+    /* inits */
+    drreg_options_t ops = { sizeof(ops), 1 /*max slots needed: aflags*/, false };
+    dr_set_client_name("DynamoRIO Sample Client 'opcode_count'",
+                       "http://dynamorio.org/issues");
+    if (!drmgr_init() || !drx_init() || drreg_init(&ops) != DRREG_SUCCESS)
+        DR_ASSERT(false);
+
+    /* register opcode event */
+    dr_register_exit_event(event_exit);
+    if (!drmgr_register_opcode_instrumentation_event(event_opcode_instruction,
+                                                     valid_opcode, NULL, NULL) ||
+        !drmgr_register_bb_instrumentation_event(NULL, event_app_instruction, NULL))
+        DR_ASSERT(false);
+
+    /* make it easy to tell, by looking at log file, which client executed */
+    dr_log(NULL, DR_LOG_ALL, 1, "Client 'opcode_count' initializing\n");
+#ifdef SHOW_RESULTS
+    /* also give notification to stderr */
+    if (dr_is_notify_on()) {
+#    ifdef WINDOWS
+        /* ask for best-effort printing to cmd window.  must be called at init. */
```

Comment:
style: capitalize.  I would also change the 6 full-line comments above to capitalize and have end punctuation: looks more professional and is the current style.

Reply:
Done.